cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/ScheduleTaskRunner.java

```diff
@@ -138,4 +140,16 @@ public void execute(ProgramId id, Map<String, String> sysArgs, Map<String, Strin
       SecurityRequestContext.setUserId(originalUserId);
     }
   }
+
+  /**
+   * Get the latest version for a program.
+   */
+  private ProgramId getLatestProgramId(ProgramId programId) {
+    String latestVersion = store.getLatest(new NamespaceId(programId.getNamespace()), programId.getApplication())
```

Comment:
can programId be null here ever? If not we need a null-check ?

Reply:
same qn

Reply:
Added a null-check and it will throw an exception if program not found